ubuntu/16.04/usr/local/share/container/baseimage-10.sh

```diff
@@ -25,3 +31,26 @@ check_development_start() {
 do_development_start() {
   :
 }
+
+do_build_user_ssh_keys() {
+  set +x
+  if [ -n "$BUILD_USER_SSH_PRIVATE_KEY" ]; then
+    echo "Setting up SSH keys for the build user"
+    (
+      umask 0200
+      mkdir -p /home/build/.ssh/
+      echo "$BUILD_USER_SSH_PRIVATE_KEY" | base64 --decode > /home/build/.ssh/id_rsa )
```

Comment:
it's to revert the umask after it's finished without assuming what the original umask was